packages/transaction-controller/src/utils.ts

```diff
@@ -45,6 +49,42 @@ export function normalizeTxParams(txParams: TransactionParams) {
   return normalizedTxParams;
 }
 
+/**
+ * Validates whether a transaction initiated by a specific 'from' address is permitted by the origin.
+ *
+ * @param permittedAddresses - The permitted accounts for the given origin.
+ * @param from - The address from which the transaction is initiated.
+ * @param selectedAddress - The currently selected Ethereum address in the wallet.
+ * @param origin - The origin or source of the transaction.
+ * @throws Throws an error if the transaction is not permitted.
+ */
+export async function validateTransactionOrigin(
```

Comment:
Since the data PR has been merged, should we move this to `validation.ts` now?

Reply:
I didn't push, but now it's moved to `validation.ts`!